src/cli/NativeMsgIface.cpp

```diff
@@ -204,13 +204,22 @@ void NativeMsgIface::handleRequest(TranslationRequest request) {
 void NativeMsgIface::handleRequest(ListRequest request)  {
     // Fetch remote models if necessary.
     if (request.includeRemote && models_.getRemoteModels().isEmpty()) {
+        std::shared_ptr<ListRequest> requestCopy(std::make_shared<ListRequest>(request));
```

Comment:
Why are we not working on the requests directly?

Reply:
The function will have finished and `request` has gone out of scope by the time one of the callbacks is called. I copied it over to a shared_ptr so it can share a single request between the two signal handlers. Previously we just copied (or moved) the request into the one signal handler.